web-app/uni-todo-app/src/components/TodoListItem.tsx

```diff
@@ -1,26 +1,55 @@
 import { ITodoEntity } from "../store/interfaces/ITodoEntity";
+import { DateTime } from 'luxon';
+import { useCallback, useMemo } from 'react';
 
 interface ITodoListItemProps {
   todoItem: ITodoEntity;
   onCompleteTodo: () => void;
   onDeleteTodo: () => void;
 }
+const formatDateTime = (date: string): string => {
+  return DateTime.fromISO(date).toLocaleString();
+};
+
+const useCalculateDaysAgo = (date: string) => {
```

Comment:
Not for this PR. In the future you can re-name this useTodoListItem and treat it as a logic functionality. That way you will be able to push the `onCompleteTodo` and `onDeleteTodo` inside this new hook and all functionality will be nice and tidy ;) 

Things will do what they are supposed to do

Reply:
Also, Kudos on the idea to add the `useCalculateDaysAgo` hook

Reply:
The suggestion to use useCallback for functions like onCompleteTodo and onDeleteTodo is also insightful, we can memoize these functions and prevent unnecessary re-creations when the component re-renders as those functions don't depend on any props or state.. Also we can encapsulate the logic within a custom hook, making it reusable across components where needed!

I accept the challenge ,
so there is gonna be a last commit soon with onCompleteTodo & onDeleteTodo as custom hooks!

Best regards , Dionisis.

Reply:
I instead made a custom hook useTodoListItem that uses the useCalculateDaysAgo custom hook & also encapsulates onCompleteTodo and onDeleteTodo inside!